miraclelinux/meta-debian
# qemu: Update version from 3.1+dfsg-8+deb10u9 to 3.1+dfsg-8+deb10u10

# Purpose of pull request
This version fixes following CVEs.
- CVE-2020-14394
- CVE-2020-17380
- CVE-2020-29130
- CVE-2021-3409
- CVE-2021-3592
- CVE-2021-3593
- CVE-2021-3594
- CVE-2021-3595
- CVE-2022-0216
- CVE-2022-1050

DLA: DLA 3362-1

# Test
## How to test
1. Build package
Build qemu and qemu-native packages.
```
$ bitbake qemu
$ bitbake qemu-native
```
2. Build core-image-weston
Add the following to local.conf and build core-image-weston image.
```
MACHINE = "qemuarm64"
DISTRO = "emlinux"
IMAGE_INSTALL_append = " qemu"
```
```
$ bitbake core-image-weston
```
3. Run runqemu command
The same source code is used for qemu and qemu-native packages.
The runqemu command runs qemu-native.
Because of these reasons, testing of this PR substituted with operation check of qemu-native.
```
$ runqemu qemuarm64
```

## Test result
Build succeeded.

### Building qemu
```
$ bitbake qemu
Loading cache: 100% |################################################################################################################################################| Time: 0:00:00
Loaded 2384 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-qemu_3.1+dfsg-8+deb10u10:754eea4398b0bbccb2f31237ad97c5c6eeaf7a57"
meta-debian-extended = "HEAD:c3eaaee8faf2c178b4aa500883f786f90aa4bc82"
meta-emlinux         = "HEAD:c0c24adb1bdb7f4e479976541314e9d9f1bb1f10"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###########################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 0 Found 0 Missed 0 Current 639 (0% match, 100% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2750 tasks of which 2750 didn't need to be rerun and all succeeded.
```

### Building qemu-native
```
$ bitbake qemu-native
Loading cache: 100% |################################################################################################################################################| Time: 0:00:00
Loaded 2384 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-qemu_3.1+dfsg-8+deb10u10:754eea4398b0bbccb2f31237ad97c5c6eeaf7a57"
meta-debian-extended = "HEAD:c3eaaee8faf2c178b4aa500883f786f90aa4bc82"
meta-emlinux         = "HEAD:c0c24adb1bdb7f4e479976541314e9d9f1bb1f10"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###########################################################################################################################################| Time: 0:00:00
Sstate summary: Wanted 0 Found 0 Missed 0 Current 35 (0% match, 100% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 335 tasks of which 335 didn't need to be rerun and all succeeded.
```

### Run runqemu command
Run runqemu command to run qemuarm64 machine.
![Screenshot from 2023-04-12 17-32-45](https://user-images.githubusercontent.com/118246107/231402257-1c5d615d-6a2c-4282-9c6f-e5956bd7598f.png)
